engine/testabilities/testmode/db_mode.go

```diff
@@ -48,5 +95,122 @@ func CheckPostgresMode() (ok bool, dbName string) {
 
 // CheckFileSQLiteMode checks if the test mode is set to use SQLite file
 func CheckFileSQLiteMode() bool {
-	return os.Getenv(modeEnvVar) == "file"
+	return os.Getenv(EnvDBMode) == "file"
+}
+
+// TestContainer represents a running test container
+type TestContainer struct {
+	Container testcontainers.Container
+	Host      string
+	Port      string
+}
+
+// GetOrCreatePostgres returns an existing PostgreSQL container or creates a new one if none exists
+// This helps reuse containers between tests
+func GetOrCreatePostgres(t testing.TB) *TestContainer {
+	t.Helper()
+
+	containerMutex.Lock()
+	defer containerMutex.Unlock()
+
+	if sharedContainer != nil {
+		t.Log("Reusing existing PostgreSQL container")
+		return sharedContainer
+	}
+
+	container := startPostgresContainer(t)
+	sharedContainer = container
+
+	if !ShouldSkipCleanup() {
+		t.Cleanup(func() {
+			containerMutex.Lock()
+			defer containerMutex.Unlock()
+
+			if sharedContainer != nil {
+				t.Log("Cleaning up PostgreSQL container")
+				ctx := context.Background()
+				if err := sharedContainer.Container.Terminate(ctx); err != nil {
+					t.Logf("Error terminating container: %v", err)
+				}
+				sharedContainer = nil
+			}
+		})
+	}
+
+	return container
+}
+
+// startPostgresContainer starts a PostgreSQL container
+func startPostgresContainer(t testing.TB) *TestContainer {
+	t.Helper()
+
+	skipCleanup := ShouldSkipCleanup()
+	if skipCleanup {
+		t.Log("Container cleanup will be skipped (EnvSkipCleanup=true)")
+	}
+
+	ctx := context.Background()
+
+	req := testcontainers.ContainerRequest{
+		Image:        "postgres:14",
+		ExposedPorts: []string{"5432/tcp"},
+		Env: map[string]string{
+			"POSTGRES_USER":     "postgres",
+			"POSTGRES_PASSWORD": "postgres",
+			"POSTGRES_DB":       "postgres",
+		},
+		WaitingFor: wait.ForAll(
+			wait.ForLog("database system is ready to accept connections").WithStartupTimeout(30*time.Second),
+			wait.ForSQL("5432/tcp", "postgres", func(host string, port nat.Port) string {
+				return fmt.Sprintf("postgres://postgres:postgres@%s:%s/postgres?sslmode=disable", host, port.Port())
+			}).WithStartupTimeout(30*time.Second),
+		),
+	}
+
+	container, err := testcontainers.GenericContainer(ctx, testcontainers.GenericContainerRequest{
+		ContainerRequest: req,
+		Started:          true,
+	})
+	if err != nil {
+		t.Fatalf("Failed to start PostgreSQL container: %v", err)
+	}
+
+	host, err := container.Host(ctx)
+	if err != nil {
+		t.Fatalf("Failed to get container host: %v", err)
+	}
+
+	mappedPort, err := container.MappedPort(ctx, nat.Port("5432/tcp"))
+	if err != nil {
+		t.Fatalf("Failed to get mapped port: %v", err)
+	}
+
+	time.Sleep(1 * time.Second)
```

Comment:
I wonder if we could get rid of this?
maybe there is some method to wait for container to be ready as we're setting wait values on container request